src/connectors/impls/gbq/writer/sink.rs

```diff
@@ -0,0 +1,660 @@
+            .mapping
+            .as_ref()
+            .ok_or(ErrorKind::BigQueryClientNotAvailable(
+                "The mapping is not available",
+            ))?;
+
+        let mut serialized_rows = vec![];
+
+        for data in event.value_iter() {
+            serialized_rows.push(mapping.map(data)?);
+        }
+
+        let request = AppendRowsRequest {
+            write_stream: write_stream.name.clone(),
+            offset: None,
+            trace_id: "".to_string(),
+            rows: Some(append_rows_request::Rows::ProtoRows(ProtoData {
+                writer_schema: Some(ProtoSchema {
+                    proto_descriptor: Some(mapping.descriptor().clone()),
+                }),
+                rows: Some(ProtoRows { serialized_rows }),
+            })),
+        };
+
+        let append_response = client
+            .append_rows(stream::iter(vec![request]))
+            .timeout(Duration::from_secs(self.config.request_timeout))
+            .await;
+
+        let append_response = if let Ok(append_response) = append_response {
+            append_response
+        } else {
+            ctx.notifier.connection_lost().await?;
+
+            return Ok(SinkReply::FAIL);
+        };
+
+        if let Ok(x) = append_response?
+            .into_inner()
+            .next()
+            .timeout(Duration::from_secs(self.config.request_timeout))
+            .await
+        {
+            match x {
+                Some(Ok(_)) => Ok(SinkReply::ACK),
+                Some(Err(e)) => {
+                    error!("BigQuery error: {}", e);
+
+                    Ok(SinkReply::FAIL)
+                }
+                None => Ok(SinkReply::NONE),
+            }
+        } else {
+            ctx.notifier.connection_lost().await?;
+
+            Ok(SinkReply::FAIL)
+        }
+    }
+
+    async fn connect(&mut self, _ctx: &SinkContext, _attempt: &Attempt) -> Result<bool> {
+        error!("Connecting to BigQuery");
+        let token = Token::new()?.header_value()?;
+
+        let token_metadata_value = MetadataValue::from_str(token.as_str())?;
+
+        let tls_config = ClientTlsConfig::new()
+            .ca_certificate(Certificate::from_pem(googapis::CERTIFICATES))
+            .domain_name("bigquerystorage.googleapis.com");
+
+        let channel = Channel::from_static("https://bigquerystorage.googleapis.com")
+            .connect_timeout(Duration::from_secs(self.config.connect_timeout))
+            .tls_config(tls_config)?
+            .connect()
+            .await?;
+
+        let mut client = BigQueryWriteClient::with_interceptor(
+            channel,
+            AuthInterceptor {
+                token: token_metadata_value,
+            },
+        );
+
+        let write_stream = client
+            .create_write_stream(CreateWriteStreamRequest {
+                parent: self.config.table_id.clone(),
+                write_stream: Some(WriteStream {
+                    name: "".to_string(),
```

Comment:
the name of the write stream should be configurable for the connector.

Reply:
Lets add a comment that for the `createWriteStream` call only the `type` is important, all other info will be filled in to the `response`. Took a while until it made `click` in my head.